pkg/scheduler/scheduler.go

```diff
@@ -399,7 +400,8 @@ func (s *Scheduler) QuerierLoop(querier schedulerpb.SchedulerForQuerier_QuerierL
 		r := req.(*queue.SchedulerRequest)
 
 		queueTime := time.Since(r.EnqueueTime)
-		s.queueDuration.Observe(queueTime.Seconds())
+		additionalQueueDimensionLabels := strings.Join(r.AdditionalQueueDimensions, ":")
```

Comment:
This there any guarantee that these are always in the same order?

Reply:
so right now we only ever send a slice of length one or nil.

In the future we would want to sort them if we stick with this approach and add any dimensions

Reply:
No, but we don't want to sort either because order does matter, as the additional dimensions represent a path through the tree.

Sending additional queue dimensions `["A" "B"]` would represent ` path `root -> tenantID -> A -> B`, which is a different queue than `["B" "A"]`, which would represent path `root -> tenantID -> B -> A`.

But for right now we send no (nil slice) additional dimensions ,or one additional dimension representing the expected query component(s) used, making the path through the tree essentially like `root -> tenantID -> queryComponent`.

The fact that we always have the first queuing dimension as tenant ID for now is also what makes me hesitant to remove  "additional" from the variable and label naming.

a path through the tree